tiledb/sm/filter/test/unit_filter_create.h

```diff
@@ -0,0 +1,36 @@
+/**
+ * @file tiledb/sm/filter/test/unit_filter_create.h
```

Comment:
If you don't anticipate needing to use a header more than once, there doesn't need to be a header. I don't see that we'd need multiple test files for `filter_create.cc`, so this header is likely unnecessary.